4_gap_analysis/src/Con_byState.R

```diff
@@ -1,95 +1,103 @@
+# Reading files locally
+#data_files <- dir('1_wqpdata/out/data', pattern=paste(state,'*', sep=''),full.names = T) %>%
+#  grep('(nitrate|phosphorus)', ., value=TRUE)
 
-# combined data may not work
+# combine N/P data into one data frame
 combined_data <- bind_rows(
   lapply(data_files, function(dfile) {
-    cname <- gsub(paste('1_wqpdata/out/data/',state,'_',sep=""), '', gsub("_00.\\.feather", '', dfile))
+    # By State
+    cname <- gsub(paste('1_wqpdata/out/data/',state,'_',sep=''), '', gsub('_00.\\.feather', '', dfile))
     dat <- feather::read_feather(dfile) %>%
-     # mutate(ConstituentGroup=cname) %>%
-    # Below here are data conversion problems
-      mutate_at(vars(starts_with("Result")),funs(as.character)) %>% # FL
-      mutate_at(vars(ends_with("HeightMeasure.MeasureValue")),funs(as.character))# AL
+      mutate(ConstituentGroup=cname) %>%
+      # Below here are data coercion fixes for columns(1 - FL, 2- AL, 3- MA, 4 - MI, 5 - PA )
+      mutate_at(vars(starts_with('Result'),ends_with('HeightMeasure.MeasureValue'),ends_with('HeightMeasure.MeasureValue'),
+                     starts_with('Measure'),starts_with('LaboratoryName'),starts_with('ActivityComment')),
+                funs(as.character))
   })
-)%>%
-  mutate(ResultSampleFractionText=ifelse(is.na(ResultSampleFractionText),"Unknown",ResultSampleFractionText))
-
-# If we are using the ActivityStartDateTime field, get rid of NAs
-# Don't understand the replacement error warning error, lens of both
-# objects are 1
-combined_data$ActivityStartDateTime[is.na(combined_data$ActivityStartDateTime)]<-
-  as.POSIXlt(paste(combined_data$ActivityStartDate[is.na(combined_data$ActivityStartDateTime)],
-                   "12:00:00",sep=" "),format="%Y-%m-%d %H:%M:%S")
-
-# Crosswalk implementation, check with Gretchen
-# Does not include Organic P (236) and Polyphosphate (55)
-# List of Agency Names, Constitutents, and Filter Type from WQ Pull
-combined_data$OrganizationIdentifier[grepl("USGS*", combined_data$OrganizationIdentifier)]<-"USGS"
-# Subset conList to only P
-conList<-filter(combined_data,grepl("*(P|p)hos*",CharacteristicName)) %>% # Check for Orthophosphate and SRP with this
-  distinct(OrganizationIdentifier,CharacteristicName,ResultSampleFractionText) %>% #6/28 removed organization Name OrganizationFormalName,
-mutate(ResultSampleFractionText=ifelse(is.na(ResultSampleFractionText),"Unknown",ResultSampleFractionText))
-
-# Crosswalk based on Harmonization table
-xWalk1<-read.csv("andy/NH2012.csv",stringsAsFactors = F)
-# Crosswalk based on Agency List and acronyms
-xWalk2<-read.csv("andy/Crosswalk.csv",stringsAsFactors = F)
+) %>% # Non-coercion changes to data, filling in NA, modifying some attributes
+   mutate(ResultSampleFractionText=ifelse(is.na(ResultSampleFractionText),'Unknown',ResultSampleFractionText)) %>%
+   # Remove nitrite samples, likely groundwater
+   filter(!CharacteristicName == 'Nitrite') %>%
```

Comment:
this probably works fine, but you could use `CharacteristicName != 'Nitrite'` instead of `!CharacteristicName == 'Nitrite'` if desired. i think it's a bit cleaner - no need to guess at order of operations.